### DISCUSSION TITLE
Configuration custom board esp32_wrover_ie_n16r8 helkp

### DISCUSSION THREAD

**Discussion by rubenmendezingenieria** [zephyrproject-rtos/zephyr#79488] [Q&A]
Hi everyone,

I am encountering difficulties while configuring my custom board with the ESP32_WROVER_IE_N16R8 microcontroller in Zephyr RTOS. Although I have prior experience working with Zephyr on other microcontrollers, this is my first time working with Espressif devices, and I have been unable to get it to work properly.

I have a few questions that I hope someone can help me with:
-In my Kconfig.kosmos200: I defined the next
config BOARD_ESP_KOSMOS200
select SOC_ESP32_WROVER_E_N16R8
select SOC_ESP32_PROCPU

what is the difference between select SOC_ESP32_PROCPU or SOC_ESP32_APPCPU? DO I have to select both?

-In my case I work with ESP32_WROVER_IE_N16R8 but I just see zephyr support to ESP32_WROVER_E_N16R8, I guess is the same since both are so similar? Do I need to have something in count?

-I am getting errors when I build with (.venv) C:\Users\Ruben\kosmos200\kosmos200-application>west -v build -b esp32_wrover_ie_n16r8_kosmos200/esp32/procpu app

CMake Error at C:/Users/Ruben/kosmos200/zephyr/cmake/modules/arch_v2.cmake:36 (message):
ARCH not defined. Check that BOARD=esp32_wrover_ie_n16r8_kosmos200, is
selecting an appropriate SoC in Kconfig, SoC=, and that the SoC is
selecting the correct architecture.

I really do not understand why since I specify in my kosmos200.yaml the arch: xtensa

I have attached my board code for your reference, and I would greatly appreciate any insights or suggestions that could help me resolve these issues.

Best regards

[boards.zip](https://github.com/user-attachments/files/17274796/boards.zip)

**Comment by rubenmendezingenieria** [comment]
solved

**Reply by moqmar** [reply]
Can you elaborate maybe? Have you figured out the difference between APPCPU & PROCPU and what to flash where? I'm also a bit stuck there :)

**Reply by rubenmendezingenieria** [reply]
sorry, I didnt see this message. I think you can find the solution on discord channel. and if you dont find it let me know and I will help you.

Best regards
